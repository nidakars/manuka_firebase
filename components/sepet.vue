<template>
  <div class="row mb">
    <div class="fl col-12 lightBg" id="pageHeader">
      <span class="col col-3 col-sm-12 pageTitle passed">
        <span class="row basket"> SEPET </span>
      </span>
      <span class="col col-3 col-sm-12 pageTitle">
        <span class="row address"> FATURA TESLİMAT </span>
      </span>
      <span class="col col-3 col-sm-12 pageTitle">
        <span class="row payment"> ÖDEME </span>
      </span>
      <span class="col col-3 col-sm-12 pageTitle">
        <span class="row confirm"> ONAY </span>
      </span>
    </div>
    <div class="box col-12 progress">
      <span class="fw600 a-center ease progress-bar" style="width: 25%"
        >25%</span
      >
    </div>
    <form
      name="SepetForm"
      id="SepetForm"
      class="box col-12 basketWrapper"
      action="Sepet.php"
      method="GET"
    >
      <div class="fl col-12">
        <div class="fl col-12 middleItem">
          <div class="col col-12 line-bottom dn-xs">
            <div class="box col-2 dn-xs text-description text-semibold">
              Ürün
            </div>
            <div
              class="box col-5 col-sm-5 col-xs-7 text-description text-semibold"
            >
              Ürün Adı
            </div>
            <div class="box col-2 text-description text-semibold">Adet</div>

            <div
              class="box col-2 text-description text-semibold text-center AdetFiyatBaslik"
            >
              Birim Fiyat
            </div>
            <div class="box col-1 text-description text-semibold text-right">
              Sil
            </div>
          </div>
          <div>Sepet Listesi</div>
          <div
            v-for="item in basket"
            :key="'basket' + item.id"
            class="box col-12 d-flex line-top productRow"
          >
            <div class="box col-2 col-sm-3 col-xs-4 productImg">
              <div class="col col-12">
                <div class="row">
                  <div class="fl image-wrapper">
                    <div class="imgInner">
                      <img :src="item.product.image" alt="item.product.title" />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div class="col col-5 col-sm-9 col-xs-8 productInfoRow">
              <span class="fl col-12 basketProName">
                <a
                  :href="'/urun-detay/' + item.product.id"
                  class="fl col-12 text-title text-bold text-upper dn-sm"
                ></a>

                <nuxt-link
                  :to="'/urun-detay/' + item.product.id"
                  class="fl col-12 col-sm-7 text-description"
                  target="_blank"
                >
                  >{{ item.product.title }} {{ item.size + 1 }}</nuxt-link
                >

                <div class="box col-sm-5 text-center AdetFiyatBox">
                  <span class="text-custom-gray">
                    {{ item.product.price }}
                  </span>
                </div>
              </span>

              <span class="fl col-12 p-bottom dn-sm">
                <span class="row"> </span>
              </span>
            </div>
            <div class="fl col-4 col-sm-9 col-xs-8 d-flex productPrcRow">
              <div
                class="box col-6 bAdetBox"
                data-product="3296"
                data-subproduct="3245"
              >
                <div class="fl col-md-12 qtyBtns" data-increment="1">
                  <!--<a title="-" data-id="0" data-cart-id="0" data-callback="qtyChange" class="col-4 d-flex decBasketProduct">
                                        <p class="fl col-12 icon-minus icon-no-space text-center p-bottom"></p>
                                    </a>-->
                  <div class="bAdet fl">Adet :</div>
                  <input
                    type="text"
                    id="Adet0"
                    name="Adet0"
                    data-cart-id="0"
                    :value="item.count"
                    class="col-4 qty"
                  />
                  <!-- <a title="+" data-id="0" data-cart-id="0" data-callback="qtyChange" class="col-4 d-flex incBasketProduct">
                                        <p class="fl col-12 icon-plus icon-no-space text-center p-bottom"></p>
                                    </a>-->
                </div>
              </div>

              <div class="box col-6 text-center AdetFiyatBox dn-sm">
                <span class="text-custom-gray">
                  {{ item.product.price }}
                </span>
              </div>
            </div>
            <div class="box col-1 col-sm-8 mSilFavori">
              <a
                class="fr btn-custom-light-gray icon-cancel icon-small icon-no-space text-center removeBtn"
                @click="removeBasketItem(item.id)"
                alt="0"
                ><span class="mSilText"></span
              ></a>
              <input type="hidden" id="cartproduct_id" value="3296" />
            </div>
          </div>

          <input type="hidden" name="AltUrun[]" id="AltUrun[]" value="" />
        </div>

        <div class="box col-12 p-bottom">
          <div class="row">
            <div class="col col-12 line-bottom line-left">
              <div class="row" id="basketBtns">
                <a
                  href="#"
                  data-width="500"
                  data-url="/srv/service/content/get/1054/general-order-note"
                  class="NotEkle popupWin btn btn-small col-sm-6 col-xs-12 btn-custom-light-gray text-center dn-sm"
                >
                  <span>Sipariş Notu Ekle</span>
                </a>

                <a
                  class="btn btn-small col-sm-6 col-xs-12 text-center btn-custom-light-gray"
                  @click="removeBasketAll()"
                >
                  Sepeti Temizle
                </a>
                <nuxt-link
                  class="btn btn-small col-sm-6 col-xs-12 text-center btn-custom-light-gray"
                  to="giyim"
                >
                  Alışverişe Devam
                </nuxt-link>
                <!-- <a if="IS_IMAGE_ADDING_ACTIVE==1" class="btn btn-small col-sm-6 col-xs-12 text-center btn-custom-light-gray popupWin" data-url="/srv/service/content/get/1054/image-notes" data-width="500">
                                    Resim Ekle
                                </a> -->

                <a
                  data-width="500"
                  data-url="/srv/service/customer/login-form"
                  class="popupWin btn btn-small col-sm-6 col-xs-12 btn-custom-light-gray text-center"
                >
                  Alışveriş Listeme Ekle
                </a>
                <nuxt-link
                  to="/sepet"
                  onclick="if ($(this).data('disabled') == 1) {
                                            return false;
                                        }
                                        $(this).data('disabled', 1);"
                  class="btn btn-small col-sm-6 col-xs-12 btn-custom-light-gray text-center"
                  title="Sepet de herhangi bir değişiklik yaptıysanız, alışverişe devam etmeden mutlaka bu butona basıp güncelleme yapmanız gerekir."
                  id="updateBasket"
                >
                  Sepeti Güncelle
                </nuxt-link>
              </div>
            </div>
            <div class="col col-3 col-md-6 col-sm-12 fr">
              <div class="row">
                <div class="col col-12">
                  <div class="row box-border b-top">
                    <div class="box col-6 line-right">Sepet Toplamı</div>
                    <div class="box col-6 a-right">{{ subtotal }} TL</div>
                  </div>

                  <!-- HOPI -->

                  <!-- HOPI -->
                  <div class="row mb box-border b-top">
                    <div class="box col-6 line-right">Genel Toplam</div>
                    <div class="box col-6 a-right">{{ total }} TL</div>
                  </div>
                </div>

                <a class="fr" id="guzellikBtn"></a>
              </div>
            </div>
            <div class="col col-9 col-md-6 col-sm-12 CuponsBox">
              <div class="row">
                <div class="box col-12 p-bottom">
                  <div class="row" id="basketOptions">
                    <div class="col col-12 p-left">
                      <div class="col col-12">
                        <div class="row mb">
                          <label
                            for="indirimkuponu"
                            class="box col-12 form-control large lightBg"
                          >
                            <span class="btn btn-medium btn-upper col-12">
                              <span class="row text-title">
                                İndirim Kuponu
                              </span>
                            </span>
                            <span class="fl col-12">
                              <span class="col col-8 p-left">
                                <input
                                  type="text"
                                  name="indirimkuponu"
                                  id="indirimkuponu"
                                  class="fl col-12"
                                  value=""
                                />
                              </span>
                              <a
                                class="btn btn-medium btn-custom-gray text-center text-upper col-4"
                                name="indirim"
                                id="indirim"
                                >Uygula</a
                              >
                            </span>
                          </label>
                        </div>
                      </div>
                      <!-- HOPI -->

                      <!-- HOPI -->
                    </div>
                    <div class="col col-6 col-sm-12 p-left">
                      <div class="col col-12 lightBg"></div>
                    </div>
                    <!-- HOPI -->

                    <!-- HOPI -->
                    <!-- ZUBIZU -->

                    <!-- /ZUBIZU -->
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="col col-12">
          <div class="row">
            <div class="fl col-12 d-flex" id="cart-footer">
              <div class="box col-9 col-md-6 col-sm-12 p-top p-left">
                <div class="col col-12 p-left">
                  <div class="fl col-12 danger" id="messages">
                    <div class="row"></div>
                  </div>
                </div>
                <p class="fl col-12">
                  Sepetinizde değişiklik yaptıysanız
                  <b>güncellemeyi unutmayın.</b> Ödeme ve onay sayfasına kadar
                  satın alma işleminizi <b>iptal edebilirsiniz.</b>
                </p>
              </div>
              <div class="col col-3 col-md-6 col-sm-12">
                <div class="row mb">
                  <a
                    href="/uye_giris"
                    class="btn btn-big col-12 btn-custom-pink text-center text-upper buyBtn"
                    title="Fatura ve adres bilgilerinizi girip alışverişinize devam etmek için bu butonu tıklayınız."
                    id="addCartBtn"
                  >
                    Satın Al
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</template>

<script>
export default {
  data: () => {
    return {}
  },
  created() {},
  computed: {
    subtotal() {
      let total = 0
      for (let i = 0; i < this.basket.length; i++) {
        total += this.basket[i].product.price
      }
      return total.toFixed(2)
    },
    total() {
      let total = 0
      for (let i = 0; i < this.basket.length; i++) {
        total += this.basket[i].product.price * this.basket[i].count
      }
      return total.toFixed(2)
    },
  },
  methods: {
    basket() {
      return this.$store.getters['basket/getBasketItems']
    },
    removeBasketItem(id) {
      this.$store.dispatch('basket/removeBasketItem', id)
    },
    removeBasketAll() {
      this.$store.dispatch('basket/removeBasketAll')
    },

    increase(id) {
      this.$store.dispatch('basket/increaseBasketItem', id)
    },
    decrease(id) {
      this.$store.dispatch('basket/decreaseBasketItem', id)
    },
  },
}
</script>
