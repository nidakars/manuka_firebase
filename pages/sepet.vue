<template>
  <div id="app">
    <Sepet></Sepet>
  </div>
</template>
<script>
import Sepet from '@/components/sepet.vue'

export default {
  name: 'sepet',
  components: {
    Sepet,
  },
}
</script>
