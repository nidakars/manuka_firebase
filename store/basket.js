export const state = () => ({
    items: [],
});
export const mutations = {
    addBasketItem(state, item) {
        state.items.push(item);
    },
    removeBasketItem(state, index) {
        state.items.splice(index, 1);
    },
    removeBasketAll(state, index) {
        state.items[index].count++;
    },
    increase(state, index) {
        state.items[index].count++;
    },
    decrease(state, index) {
        state.items[index].count--;
    }
};
export const actions = {
    addBasket({ state, commit }, data) {
        for (let i = 0; i < state.items.length; i++) {
            if (state.items[i].product.id === data.product.id && state.items[i].size === data.size) {
                commit('increase', i);
                return true;
            }
        }
        commit('addBasketItem', data);
    },
    increaseBasketItem({ state, commit }, id) {
        for (let i = 0; i < state.items.length; i++) {
            if (state.items[i].id === id) {
                commit('increase', i);
                return false;
            }
        }
    },
    decreaseBasketItem({ state, commit }, id) {
        for (let i = 0; i < state.items.length; i++) {
            if (state.items[i].id === id) {
                if (state.items[i].count < 2) {
                    commit('removeBasketItem', i);
                } else {
                    commit('decrease', i);
                }
                return false;
            }
        }
    },
    removeBasketItem({ state, commit }, id) {
        for (let i = 0; i < state.items.length; i++) {
            if (state.items[i].id === id) {
                commit('removeBasketItem', i);
                return false;
            }
        }
    },
    removeBasketAll({ state, commit }) {
        for (let i = 0; i < state.items.length; i++) {
            commit('removeBasketItem', i);

        }
    }
};
export const getters = {
    getBasketItems(state) {
        return state.items;
    }
};