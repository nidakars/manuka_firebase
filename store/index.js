export const mutations = {
    setProducts(state, param) {
        state.products = param;
    },
    setProduct(state, product) {
        state.product = product;
    },
    addBasketItem(state, item) {
        state.items.push(item);
    },
    removeBasketItem(state, index) {
        state.items.splice(index, 1);
    },
    removeBasketAll(state, index) {
        state.items[index].count++;
    },
    increase(state, index) {
        state.items[index].count++;
    },
    decrease(state, index) {
        state.items[index].count--;
    }
};
export const actions = {
    SetProduct({ state, commit }, id) {
        for (let i = 0; i < state.products.length; i++) {
            if (state.products[i].id === id) {
                commit("setProduct", state.products[i]);
            }
        }
    },
    fetchProducts({ commit }) {
        return new Promise((resolve, reject) => {
            //Düzeltme 5
            realDb.ref('/products').once('value').then((snapshot) => {

                //console.log(snapshot.val());
                let arr = []
                if (snapshot.val() != null) {
                    //Map tipinde bir değişkeni arraye dönüştürmek için kullandığımız kod:
                    arr = Object.entries(snapshot.val()).map(e => Object.assign(e[1], { key: e[0] }))
                }
                commit('setProducts', arr)

            });
        })
    },
    async logout({}) {
        try {
            await firebase
                .auth()
                .signOut()
                .then(() => {
                    //this.$router.push("/login/signin");
                    location.reload();
                });
        } catch (err) {
            console.log(err);
        }
    },
    register({}, payload) {
        firebase
            .auth()
            .createUserWithEmailAndPassword(payload.email, payload.password)
            .then(
                userCredential => {
                    // Kayıt işlemi sadece email ve şifre ile oluşturuluyor, baştan isim ve numara ekleyemiyoruz.
                    alert("Hesap " + payload.email + " başarıyla oluşturuldu");
                    // Email ve şifre ile kullanıcı oluşturduktan sonra hesabın isim ve numarasını güncelliyoruz.
                    var kullanici = firebase.auth().currentUser;
                    alert("Kullanıcıya giriş yapıldı");
                    kullanici.updateProfile({
                        displayName: payload.name,
                        phoneNumber: payload.phone
                    });
                    // alert("İsim " + payload.name + " güncellendi");
                    // alert("Telefon " + payload.phone + " güncellendi");
                    var newUser = userCredential.user
                    alert("oluşturulan kullanıcı idsi: " + newUser.uid)

                    var ref = realDb.ref("usersData/" + newUser.uid) //.set(userData)
                    var uData = {
                        adresses: null,



                        id: newUser.uid,


                        ordersId: null
                    }
                    alert("Kullanıcı verisi Realtime'a kaydedildi")
                    ref.set(uData);
                    //payload.$router.push("/"); 
                    location.reload();
                },
                err => {
                    alert(err.message);
                }
            ).then(

            );
    },
    login({}, payload) {
        console.log("login çalıştı.")
        console.log(payload.email)
        try {
            firebase
                .auth()
                .signInWithEmailAndPassword(payload.email, payload.password)
                .then(
                    user => {
                        alert(payload.email + " ile giriş yaptınız");
                        //this.$router.push("/");
                        location.reload();
                    },
                    err => {
                        alert(err.message);
                    }
                );
        } catch (error) {
            console.log(error)
        }
    },
    updateUser({}, payload) {},
    addBasket({ state, commit }, data) {
        for (let i = 0; i < state.items.length; i++) {
            if (state.items[i].product.id === data.product.id && state.items[i].size === data.size) {
                commit('increase', i);
                return true;
            }
        }
        commit('addBasketItem', data);
    },
    increaseBasketItem({ state, commit }, id) {
        for (let i = 0; i < state.items.length; i++) {
            if (state.items[i].id === id) {
                commit('increase', i);
                return false;
            }
        }
    },
    decreaseBasketItem({ state, commit }, id) {
        for (let i = 0; i < state.items.length; i++) {
            if (state.items[i].id === id) {
                if (state.items[i].count < 2) {
                    commit('removeBasketItem', i);
                } else {
                    commit('decrease', i);
                }
                return false;
            }
        }
    },
    removeBasketItem({ state, commit }, id) {
        for (let i = 0; i < state.items.length; i++) {
            if (state.items[i].id === id) {
                commit('removeBasketItem', i);
                return false;
            }
        }
    },
    removeBasketAll({ state, commit }) {
        for (let i = 0; i < state.items.length; i++) {
            commit('removeBasketItem', i);
        }
    }
};
//import { realDb } from '~/plugins/firebase.js';
export const state = () => ({
    isInitialized: false,
    products: [],
    product: null,
    items: [],
    test: "Bu bir test mesajıdır."
});

// export const state = () => ({
//     isInitialized: false,
//     user: null
// });
// export const mutations = {
//     setUser(state, param) {
//         state.user = param;
//     }
// };
// export const actions = {};
// export const getters = {
//     getUser(state) {
//         return state.user;
//     }
// };
/////////////////////////////////////////////////////////////

import firebase from "firebase";
import { realDb, auth } from '../plugins/firebase.js';

export const getters = {
    getBasketItems(state) {
        return state.items;
    },
    getProducts: (state) => () => {
        return state.products;
    },
    getProduct: (state) => (id) => {
        return state.products.find(product => product.id == id);
    },
    getTest: (state) => () => {
        return state.test
    },
    isLogged: () => () => {
        firebase.auth().onAuthStateChanged(user => {
            return user ? true : false;
        });
    },
    getUserdata: () => () => {
        // Kullanıcı verilerini çekiyoruz.
        if (this.isLoggedIn) {
            // kullanici diye bir variable tanımladık. Email ile ismi çektik.
            var kullanici = firebase.auth().currentUser;
            return {
                email: kullanici.email,
                name: kullanici.displayName,
                uid: kullanici.uid,
            }
        }
    }
}