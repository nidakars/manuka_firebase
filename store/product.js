import { realDb } from '~/plugins/firebase.js';
export const state = () => ({
    isInitialized: false,
    products: [],
    product: null
});
export const mutations = {
    setProducts(state, param) {
        state.products = param;
    },
    setProduct(state, product) {
        state.product = product;
    }
};
export const actions = {
    // initData({ state, commit }, ) {
    //     if (state.isInitialized === false) {
    //         commit("setProducts", data);
    //         state.isInitialized = true;
    //     }
    // },
    SetProduct({ state, commit }, id) {
        for (let i = 0; i < state.products.length; i++) {
            if (state.products[i].id === id) {
                commit("setProduct", state.products[i]);
            }
        }
    },
    fetchProducts({ commit }) {
        return new Promise((resolve, reject) => {
            //Düzeltme 5
            realDb.ref('/products').once('value').then((snapshot) => {

                //console.log(snapshot.val());
                let arr = []
                if (snapshot.val() != null) {
                    //Map tipinde bir değişkeni arraye dönüştürmek için kullandığımız kod:
                    arr = Object.entries(snapshot.val()).map(e => Object.assign(e[1], { key: e[0] }))
                }
                commit('setProducts', arr)

            });
        })
    },
};
export const getters = {
    getProducts: (state) => () => {
        return state.products;
    },
    getProduct: (state) => (id) => {
        return state.products.find(product=>product.id==id);
    }

};