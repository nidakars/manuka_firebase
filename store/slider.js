import { realDb } from '~/plugins/firebase.js';
export const state = () => ({
    isInitialized: false,
    pics: [],
    pic: null
});
export const mutations = {
    setSliders(state, param) {
        state.pics = param;
    },
    setSlider(state, pic) {
        state.pic = pic;
    }
};
export const getters = {
    getSliders: (state) => () => {
        return state.pics;
    },
    getSlider: (state) => (id) => {
        return state.pics.find(pic => pic.id == id);
    }

};
export const actions = {
    SetSlider({ state, commit }, id) {
        for (let i = 0; i < state.pics.length; i++) {
            if (state.pics[i].id === id) {
                commit("setSlider", state.pics[i]);
            }
        }
    },
    fetchSliders({ commit }) {
        return new Promise((resolve, reject) => {
            //Düzeltme 5
            realDb.ref('/slider').once('value').then((snapshot) => {

                //console.log(snapshot.val());
                let arr = []
                if (snapshot.val() != null) {
                    //Map tipinde bir değişkeni arraye dönüştürmek için kullandığımız kod:
                    arr = Object.entries(snapshot.val()).map(e => Object.assign(e[1], { key: e[0] }))
                }
                commit('setSliders', arr)

            });
        })
    },
};