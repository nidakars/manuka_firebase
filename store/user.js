// export const state = () => ({
//     isInitialized: false,
//     user: null
// });
// export const mutations = {
//     setUser(state, param) {
//         state.user = param;
//     }
// };
// export const actions = {};
// export const getters = {
//     getUser(state) {
//         return state.user;
//     }
// };
/////////////////////////////////////////////////////////////

import firebase from "firebase";
import { realDb, auth } from '~/plugins/firebase.js';
export const state = () => ({
    test: "Bu bir test mesajıdır."
})
export const getters = {
    getTest: (state) => () => {
        return state.test
    },
    isLogged: () => () => {
        firebase.auth().onAuthStateChanged(user => {
            return user ? true : false;
        });
    },
    getUserdata: () => () => {
        // Kullanıcı verilerini çekiyoruz.
        if (this.isLoggedIn) {
            // kullanici diye bir variable tanımladık. Email ile ismi çektik.
            var kullanici = firebase.auth().currentUser;
            return {
                email: kullanici.email,
                name: kullanici.displayName,
                uid: kullanici.uid,
            }
        }
    }
}
export const actions = {
    async logout({}) {
        try {
            await firebase
                .auth()
                .signOut()
                .then(() => {
                    //this.$router.push("/login/signin");
                    location.reload();
                });
        } catch (err) {
            console.log(err);
        }
    },
    register({}, payload) {
        firebase
            .auth()
            .createUserWithEmailAndPassword(payload.email, payload.password)
            .then(
                userCredential => {
                    // Kayıt işlemi sadece email ve şifre ile oluşturuluyor, baştan isim ve numara ekleyemiyoruz.
                    alert("Hesap " + payload.email + " başarıyla oluşturuldu");
                    // Email ve şifre ile kullanıcı oluşturduktan sonra hesabın isim ve numarasını güncelliyoruz.
                    var kullanici = firebase.auth().currentUser;
                    alert("Kullanıcıya giriş yapıldı");
                    kullanici.updateProfile({
                        displayName: payload.name,
                        phoneNumber: payload.phone
                    });
                    alert("İsim " + payload.name + " güncellendi");
                    alert("Telefon " + payload.phone + " güncellendi");
                    var newUser = userCredential.user
                    alert("oluşturulan kullanıcı idsi: " + newUser.uid)

                    var ref = realDb.ref("usersData/" + newUser.uid) //.set(userData)
                    var uData = {
                        adresses: null,



                        id: newUser.uid,


                        ordersId: null
                    }
                    alert("Kullanıcı verisi Realtime'a kaydedildi")
                    ref.set(uData);
                    //payload.$router.push("/"); 
                    location.reload();
                },
                err => {
                    alert(err.message);
                }
            ).then(

            );
    },
    login({}, payload) {
        console.log("login çalıştı.")
        console.log(payload.email)
        try {
            firebase
                .auth()
                .signInWithEmailAndPassword(payload.email, payload.password)
                .then(
                    user => {
                        alert(payload.email + " ile giriş yaptınız");
                        //this.$router.push("/");
                        location.reload();
                    },
                    err => {
                        alert(err.message);
                    }
                );
        } catch (error) {
            console.log(error)
        }
    },
    updateUser({}, payload) {},
}
export const mutations = {}