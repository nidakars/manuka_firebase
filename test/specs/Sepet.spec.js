import { mount, createLocalVue, config, shallowMount, RouterLinkStub } from "@vue/test-utils";
import sepet from "../../components/sepet.vue";
import { getters, mutations, actions, state } from "../../store/basket";

config.mocks["$store"] = {
    state: {...state },
    getters: {
        "basket/getBasketItems": {...state }
    },
    actions: {
        "basket/removeBasketItem": {...actions }
    },
    mutations: {
        ...mutations
    }
};

describe("sepet.vue", () => {
    let wrapper;
    wrapper = shallowMount(sepet, {
            stubs: { NuxtLink: RouterLinkStub }
        }

    );
    describe('Sepet Listesi', () => {
        it("Render", () => {
            expect(wrapper.text()).toContain("Sepet Listesi");
        });
        test("snapshot3", () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});