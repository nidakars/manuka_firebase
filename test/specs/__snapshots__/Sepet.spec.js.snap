// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sepet.vue Sepet Listesi snapshot3 1`] = `
<div class="row mb">
  <div id="pageHeader" class="fl col-12 lightBg"><span class="col col-3 col-sm-12 pageTitle passed"><span class="row basket"> SEPET </span></span> <span class="col col-3 col-sm-12 pageTitle"><span class="row address"> FATURA TESLİMAT </span></span> <span class="col col-3 col-sm-12 pageTitle"><span class="row payment"> ÖDEME </span></span> <span class="col col-3 col-sm-12 pageTitle"><span class="row confirm"> ONAY </span></span></div>
  <div class="box col-12 progress"><span class="fw600 a-center ease progress-bar" style="width: 25%;">25%</span></div>
  <form name="SepetForm" id="SepetForm" action="Sepet.php" method="GET" class="box col-12 basketWrapper">
    <div class="fl col-12">
      <div class="fl col-12 middleItem">
        <div class="col col-12 line-bottom dn-xs">
          <div class="box col-2 dn-xs text-description text-semibold">
            Ürün
          </div>
          <div class="box col-5 col-sm-5 col-xs-7 text-description text-semibold">
            Ürün Adı
          </div>
          <div class="box col-2 text-description text-semibold">Adet</div>
          <div class="box col-2 text-description text-semibold text-center AdetFiyatBaslik">
            Birim Fiyat
          </div>
          <div class="box col-1 text-description text-semibold text-right">
            Sil
          </div>
        </div>
        <div>Sepet Listesi</div> <input type="hidden" name="AltUrun[]" id="AltUrun[]" value="">
      </div>
      <div class="box col-12 p-bottom">
        <div class="row">
          <div class="col col-12 line-bottom line-left">
            <div id="basketBtns" class="row"><a href="#" data-width="500" data-url="/srv/service/content/get/1054/general-order-note" class="NotEkle popupWin btn btn-small col-sm-6 col-xs-12 btn-custom-light-gray text-center dn-sm"><span>Sipariş Notu Ekle</span></a> <a class="btn btn-small col-sm-6 col-xs-12 text-center btn-custom-light-gray">
                Sepeti Temizle
              </a> <a class="btn btn-small col-sm-6 col-xs-12 text-center btn-custom-light-gray">
                Alışverişe Devam
              </a> <a data-width="500" data-url="/srv/service/customer/login-form" class="popupWin btn btn-small col-sm-6 col-xs-12 btn-custom-light-gray text-center">
                Alışveriş Listeme Ekle
              </a> <a onclick="if ($(this).data('disabled') == 1) {
                                          return false;
                                      }
                                      $(this).data('disabled', 1);" title="Sepet de herhangi bir değişiklik yaptıysanız, alışverişe devam etmeden mutlaka bu butona basıp güncelleme yapmanız gerekir." id="updateBasket" class="btn btn-small col-sm-6 col-xs-12 btn-custom-light-gray text-center">
                Sepeti Güncelle
              </a></div>
          </div>
          <div class="col col-3 col-md-6 col-sm-12 fr">
            <div class="row">
              <div class="col col-12">
                <div class="row box-border b-top">
                  <div class="box col-6 line-right">Sepet Toplamı</div>
                  <div class="box col-6 a-right">0.00 TL</div>
                </div>
                <div class="row mb box-border b-top">
                  <div class="box col-6 line-right">Genel Toplam</div>
                  <div class="box col-6 a-right">0.00 TL</div>
                </div>
              </div> <a id="guzellikBtn" class="fr"></a>
            </div>
          </div>
          <div class="col col-9 col-md-6 col-sm-12 CuponsBox">
            <div class="row">
              <div class="box col-12 p-bottom">
                <div id="basketOptions" class="row">
                  <div class="col col-12 p-left">
                    <div class="col col-12">
                      <div class="row mb"><label for="indirimkuponu" class="box col-12 form-control large lightBg"><span class="btn btn-medium btn-upper col-12"><span class="row text-title">
                              İndirim Kuponu
                            </span></span> <span class="fl col-12"><span class="col col-8 p-left"><input type="text" name="indirimkuponu" id="indirimkuponu" value="" class="fl col-12"></span> <a name="indirim" id="indirim" class="btn btn-medium btn-custom-gray text-center text-upper col-4">Uygula</a></span></label></div>
                    </div>
                  </div>
                  <div class="col col-6 col-sm-12 p-left">
                    <div class="col col-12 lightBg"></div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="col col-12">
        <div class="row">
          <div id="cart-footer" class="fl col-12 d-flex">
            <div class="box col-9 col-md-6 col-sm-12 p-top p-left">
              <div class="col col-12 p-left">
                <div id="messages" class="fl col-12 danger">
                  <div class="row"></div>
                </div>
              </div>
              <p class="fl col-12">
                Sepetinizde değişiklik yaptıysanız
                <b>güncellemeyi unutmayın.</b> Ödeme ve onay sayfasına kadar
                satın alma işleminizi <b>iptal edebilirsiniz.</b>
              </p>
            </div>
            <div class="col col-3 col-md-6 col-sm-12">
              <div class="row mb"><a href="/uye_giris" title="Fatura ve adres bilgilerinizi girip alışverişinize devam etmek için bu butonu tıklayınız." id="addCartBtn" class="btn btn-big col-12 btn-custom-pink text-center text-upper buyBtn">
                  Satın Al
                </a></div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
